scripts/test_cases/msbuild_performance_test_linux.py

```diff
@@ -0,0 +1,171 @@
+"""_summary_
+This script basically runs benchmark test for msbuild
+Returns:
+    None: .....
+"""
+
+import subprocess
+import time
+import os
+import urllib.request
+
+EXTRACT_PATH = "sdk"
+WORKING_DIR = "msbuild-performance-test"
+DOTNET_BASE_VERSION_URL_LINUX = "https://download.visualstudio.microsoft.com/download/pr/dc930bff-ef3d-4f6f-8799-6eb60390f5b4/1efee2a8ea0180c94aff8f15eb3af981/dotnet-sdk-6.0.300-linux-x64.tar.gz"
+DOTNET_DAILY_VERSION_URL_LINUX = "https://aka.ms/dotnet/8.0.1xx/daily/dotnet-sdk-linux-x64.tar.gz"
+TEST_SOLUTION_REPO_URL = "https://github.com/marcin-krystianc/TestSolutions.git"
+TEST_REPO_NAME = "TestSolutions"
+TEST_SOLUTION_CASE = "LargeAppWithPrivatePackagesCentralisedNGBVRemoved"
+TEST_SOLUTION_DIR = "solution"
+
+
+def check_directory(event):
+    """Checking directory for debugging purposes"""
+    print(f'_____CHECKING DIR {event}________')
+    subprocess.call("ls", shell=True)
+
+
+def create_extract_destinations():
+    """ Create the extract destination directories if they do not exist"""
+    check_directory('BEFORE CREATE EXTRACT DESTINATION')
+    if not os.path.exists(EXTRACT_PATH):
+        os.mkdir(EXTRACT_PATH)
+    os.chdir(EXTRACT_PATH)
+    if not os.path.exists("base"):
+        os.mkdir("base")
+    if not os.path.exists("daily"):
+        os.mkdir("daily")
+    check_directory('AFTER CREATE EXTRACT DESTINATION')
+
+
+def download_file(url, filename):
+    """ Download file from url and save it to filename"""
+    check_directory('BEFORE DOWNLOAD FILE')
+    with urllib.request.urlopen(url) as response, open(filename, 'wb') as out_file:
+        data = response.read()
+        out_file.write(data)
+    check_directory('AFTER DOWNLOAD FILE')
+
+
+def download_and_extract_dotnet_sdk(version_url, is_base):
+    """ Download and extract the dotnet sdk"""
+    path = "base" if is_base else "daily"
+    tar_gz_file = "dotnet-sdk.tar.gz"
+    download_file(version_url, tar_gz_file)
+
+    # Extract the tar.gz file
+    check_directory('BEFORE EXTRACT FILE')
+    extract_command = f"tar -xzf {tar_gz_file} -C {path}"
+    subprocess.call(extract_command, shell=True)
+
+
+def measure_execution_time(command):
+    """summary for measure_execution_time
+
+    Args:
+        command ([String]): [command to be executed]
+
+    Returns:
+        [Number]: [ellapsed time in seconds]
+    """
+
+    check_directory('BEFORE MEASURE EXECUTION TIME')
+    subprocess.call(f"cd {TEST_REPO_NAME}", shell=True)
+    check_directory('AFTER CHECKINT OUT TO REPO')
+
+    # Record start time
+    start_time = time.time()
+
+    # Run the command
+    subprocess.call(command, shell=True)
+
+    # Calculate elapsed time
+    end_time = time.time()
+    elapsed_time = end_time - start_time
+
+    return elapsed_time
+
+
+def clone_repository(repo_url, repo_path):
```

Comment:
This method does too many things, it clones the repository and changes the current directory to the test solution dir.
The method's name suggests that it clones the repo, so it should do only that.

PS:
Generally, it is better to avoid changing the current directory because it makes it harder to understand how the program works.